code/_onclick/hud/meter/meter.dm

```diff
@@ -51,74 +50,26 @@
 
 	var/margin = 8//Extra length that isn't counted as part of our length for the purpose of components
 
-/obj/screen/meter/New(var/atom/holder)
-
+/obj/screen/meter/New(atom/holder)
+	L = holder
 	cache_data(arglist(args))
-	.=..()
 
+	.=..()
 
 //Override this and change the parameters in subtypes
-/obj/screen/meter/proc/cache_data(var/atom/holder)
+/obj/screen/meter/proc/cache_data(atom/holder)
 
 
 /obj/screen/meter/Destroy()
 	QDEL_NULL(remaining_meter)
 	QDEL_NULL(delta_meter)
 	QDEL_NULL(limit_meter)
 	QDEL_NULL(textholder)
-	if (C)
-		C.screen -= src
-		C = null
 	.=..()
 
-/obj/screen/meter/added_to_screen(var/client/newclient)
-	recreate_components(newclient)
-
-
-
-
-/obj/screen/meter/proc/recreate_components(var/client/newclient)
```

Comment:
so was this, i think most of the changes to meters will need reverted

Reply:
> so was this, i think most of the changes to meters will need reverted

Nope